src/foam/nanos/boot/NSpecFactory.java

```diff
@@ -45,10 +45,11 @@ public synchronized Object create(X x) {
 
     try {
       if ( logger != null ) logger.info("Creating Service", spec_.getName());
+      X serviceX = spec_.getName().endsWith("DAO") ? new InvalidX(x_.getX(), spec_.getName(), logger) : x_.getX();
```

Comment:
Now that I'm reviewing this, it would be better to move this below line 50 and change it like so:

```suggestion
      X serviceX = ns instanceof DAO ? new InvalidX(x_.getX(), spec_.getName(), logger) : x_.getX();
```